javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -652,7 +662,11 @@ private Vector(T[] leading, byte leadingOffset, VectorTree<T> middle, T[] traili
         if (isEmpty()) {
             return of(element);
         } else {
-            if (trailingLength < branchingFactor()) {
+            if ((trailingLength == 0) && (leadingLength() < branchingFactor())) {
+                assert (middle.length() == 0) && (trailingLength == 0);
```

Comment:
if we don't have a `trailing` yet (just a `leading`), we should append there instead